CI/package-deps-windows.ps1

```diff
@@ -0,0 +1,159 @@
+Param(
+    [Switch]$Help = $(if (Test-Path variable:Help) { $Help }),
+    [Switch]$Quiet = $(if (Test-Path variable:Quiet) { $Quiet }),
+    [Switch]$Verbose = $(if (Test-Path variable:Verbose) { $Verbose }),
+    [String]$BuildDirectory = "build"
+)
+
+##############################################################################
+# Windows OBS deps package script
+##############################################################################
+#
+# This script contains all steps necessary to:
+#
+#   * Build OBS required dependencies
+#   * Create 64-bit and 32-bit variants
+#
+# Parameters:
+#   -Help                   : Print usage help
+#   -Quiet                  : Suppress most build process output
+#   -Verbose                : Enable more verbose build process output
+#   -BuildDirectory         : Directory to use for builds
+#                             Default: Win64 on 64-bit systems
+#                                      Win32 on 32-bit systems
+#
+##############################################################################
+
+$ErrorActionPreference = "Stop"
+
+function Package-OBS-Deps-Main {
+    $CheckoutDir = git rev-parse --show-toplevel
+    $ProductName = "windows-deps"
+    $DepsBuildDir = "${CheckoutDir}\windows\obs-win-deps"
+
+    . "${CheckoutDir}\CI\include\build_support_windows.ps1"
+
+    Write-Status "Create Windows OBS deps package"
+
+    Ensure-Directory "${CheckoutDir}"
+
+    if (!$CurrentDate) {
+        $CurrentDate = Get-Date -UFormat "%Y-%m-%d"
+    }
+    $FileName = "${ProductName}-${CurrentDate}.tar.gz"
+    $CrossDirBase = "${CheckoutDir}\windows\obs-cross-deps"
+    $NativeDirBase = ""
+
+    if (Test-Path "${CheckoutDir}\windows\obs-native-deps" -PathType "Container") {
+        $NativeDirBase = "${CheckoutDir}\windows\obs-native-deps"
+    } elseif (Test-Path "${CheckoutDir}\windows_native_build_temp" -PathType "Container") {
+        $NativeDirBase = "${CheckoutDir}\windows_native_build_temp"
+    }
+
+    if (!(Test-Path "${CrossDirBase}\x86" -PathType "Container")) {
+        Caught-Error "Missing cross-compiled build in ${CrossDirBase}\x86"
+    } elseif (!(Test-Path "${CrossDirBase}\x86_64" -PathType "Container")) {
+        Caught-Error "Missing cross-compiled build in ${CrossDirBase}\x86_64"
+    } elseif (!(Test-Path "${NativeDirBase}\win32" -PathType "Container")) {
+        Caught-Error "Missing native build in ${NativeDirBase}\win32"
+    } elseif (!(Test-Path "${NativeDirBase}\win64" -PathType "Container")) {
+        Caught-Error "Missing native build in ${NativeDirBase}\win64"
+    }
+
+    Remove-ItemIfExists "${DepsBuildDir}"
+    Ensure-Directory "${DepsBuildDir}\win32"
+    Ensure-Directory "${DepsBuildDir}\win64"
+    cd ..
```

Comment:
In this case, I'm changing back to "${DepsBuildDir}", and I believe that's a full path, so I could certainly just use that directly since it is already stored as a variable, and I think that's the simplest change to make here.  I think I was being lazy because I know that I've only just created a directory one level down, and in this case it indicates the hierarchy/location clearly enough since it's a single traversal (one `..`) rather than several.

I'm not sure about whether or not it's safer.  I'm pretty sure an immediate parent directory must exist for any path except root.  Could you perhaps elaborate?

Reply:
In that particular case it should be fine, I've been bitten by functions/calls that change the working directory and so the state upon returning was not as expected.

Reply:
Ah, perhaps you mean that the working directory is changed _in general_ away from where the user may have started?  In that case, yes, the script should be a nice player and return the shell's working directory to where the user started.  As this is a pretty minor change and it's in the packaging script, I may make that now instead of pushing it till later.